common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java

```diff
@@ -293,4 +294,45 @@ public static String getJsonPath(String jsonPathWithSource) {
         // Extract the substring from the startIndex to the end of the input string
         return (startIndex != -1) ? jsonPathWithSource.substring(startIndex) : jsonPathWithSource;
     }
+
+    /**
+     * Checks if the given input string matches the dot path format.
+     *
+     * <p>The dot path format is a string that consists of one or more word characters
+     * (letters, digits, or underscores) separated by dots. The string can optionally
+     * start or end with a dot followed by one or more word characters.
+     *
+     * <p>Examples of valid dot path format strings:
+     * <ul>
+     *     <li>"foo"</li>
+     *     <li>"foo.bar"</li>
+     *     <li>"foo.bar.baz"</li>
+     *     <li>"foo.bar.baz.qux"</li>
+     *     <li>".foo"</li>
+     *     <li>"foo."</li>
+     *     <li>".foo.bar"</li>
+     * </ul>
+     *
+     * <p>Examples of invalid dot path format strings:
+     * <ul>
+     *     <li>"foo..bar"</li>
+     *     <li>"."</li>
+     *     <li>".."</li>
+     * </ul>
+     *
+     * @param input the input string to be checked
+     * @return true if the input string matches the dot path format, false otherwise
+     */
+    public static boolean isValidJSONPath(String input) {
+        if (input == null || input.equals("")) {
```

Comment:
what if `" "`? may be we can use `input.isBlank()`?

Reply:
good catch, changed to input.isBlank()